### DISCUSSION TITLE
I have an Android question.

### DISCUSSION THREAD

**Discussion by mjxaone** [rustdesk/rustdesk#619] [Q&A] [answered]
hello. I'm using rustDesk for the first time today.
I tried connecting to Android.

1. What should I do when the screen is off?
2. Is there an Android Reboot command?

**Comment by rustdesk** [comment, marked as answer]
I do not think there are apis allowing us do these. Correct me if I am wrong.

**Reply by mjxaone** [reply]
oh. ok. i understand. Thanks for your kind reply.

RustDesk approached it the normal way.
I thought that RustDesk was implemented in the same way as the STF project.
I'll take a closer look at RustDesk.

ref:
https://github.com/DeviceFarmer/stf

**Reply by rustdesk** [reply]
> stf

Can it implement above without root?

**Reply by rustdesk** [reply]
It is adb, not suitable for our scenario, we target remote controlling.

**Reply by mjxaone** [reply]
Yes. As far as I know, it is implemented without root privileges.
Line 6 of the link.
https://github.com/DeviceFarmer/stf#features

But I don't know the implementation details.
I haven't looked closely at the source of the STF.

hm.. STF may have used adb .
If so, it cannot be implemented in RustDesk unless there is a USB connection.

**Reply by rustdesk** [reply]
what else?

**Reply by mjxaone** [reply]
A second post came up while I was replying.
you're right. If you used adb , you can't implement it in RustDesk.

**Reply by mjxaone** [reply]
@hello-smile6 me?

I got to know RustDesk less than an hour ago.
I was curious so I posted a question.
I am studying RustDesk. I am very interested.

**Reply by mjxaone** [reply]
@rustdesk 
Thank you for your interest in the question and for answering it.
Many questions have been answered. :D

**Reply by rustdesk** [reply]
> studying

Wish you have a good studying journey.

**Reply by mjxaone** [reply]
@hello-smile6
Does VNC have a Relay Server?

RustDesk's Relay Server is a very interesting and important feature.

**Reply by hello-smile6** [reply]
> @hello-smile6 Does VNC have a Relay Server?
> 
> RustDesk's Relay Server is a very interesting and important feature.

Yes. **Run it yourself.**

**Reply by mjxaone** [reply]
@hello-smile6 wow...
I know about VNC Proxy.
I am hearing about VNC Relay Server for the first time.
Any solution?

Can I configure it like below?
Android Host(3G IP(NAT IP)) <-> (VNC Relay Server) <-> VNC Client

Please let me know which Application I can use.
I couldn't find any related material when I searched.

**Reply by hello-smile6** [reply]
@mjxaone Use NoVNC or ngrok.

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/wiki/FAQ#android-start-on-boot-and-limitations

any further discussion please go to https://github.com/rustdesk/rustdesk/discussions/1330
